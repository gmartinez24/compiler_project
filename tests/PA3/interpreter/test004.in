false 6.7