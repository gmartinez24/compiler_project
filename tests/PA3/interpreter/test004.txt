// Boolean expressions in nested if statement and early return statement

main

bool a, b;
int x, y; 
float z;

{
    a = call readBool();
    b = true;
    
    x = 14;
    y = 5;
    z = call readFloat();

    if (not (a and b) and (z < x)) then
        if (z >= y) then
            call printFloat(z);
        fi;
    else 
        return;
    fi;

    call printInt(x + y);
}.
