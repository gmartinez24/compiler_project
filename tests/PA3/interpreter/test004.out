Error type-checking file.
TypeError(17,30)[Cannot compare float with int.]
TypeError(17,23)[Cannot compute bool and ErrorType(Cannot compare float with int.).]
TypeError(17,9)[Cannot negate ErrorType(Cannot compute bool and ErrorType(Cannot compare float with int.).).]
TypeError(17,5)[IfStat requires bool condition not ErrorType(Cannot negate ErrorType(Cannot compute bool and ErrorType(Cannot compare float with int.).).).]
TypeError(18,15)[Cannot compare float with int.]
TypeError(18,9)[IfStat requires bool condition not ErrorType(Cannot compare float with int.).]

